### DISCUSSION TITLE
Add `NonCopyable`

### DISCUSSION THREAD

**Discussion by JackyWoo** [pocoproject/poco#3785] [General]
**Is your feature request related to a problem? Please describe.**
Providing a convenient fashion to diable copy operator of class.

**Describe the solution you'd like**
Add `NonCopyable` class who disable  copy operators and others can extend it.

**Comment by aleks-f** [comment]
This is to trivial to be included on the library level - just delete the copy constructor/assignment operator (or make it private) and you've got it.

**Reply by JackyWoo** [reply]
👌🏻
